File: Motocitizen/src/motocitizen/content/VolunteerStatus.java
package motocitizen.content;

public enum VolunteerStatus {
    ON_WAY("onway", "Выехал"),
    IN_PLACE("inplace", "На месте"),
    LEAVE("leave", "Уехал");

    private final String code;
    private final String text;

    VolunteerStatus(String code, String text) {
        this.text = text;
        this.code = code;
    }

    public static VolunteerStatus parse(String medicine) {
        for (VolunteerStatus a : VolunteerStatus.values()) {
            if (a.code.equals(medicine)) return a;
        }
        return VolunteerStatus.ON_WAY;
    }

    public String code() {
        return this.code;
    }

    public String string() {
        return this.text;
    }
}

File: Motocitizen/src/motocitizen/dictionary/VolunteerStatus.kt
package motocitizen.dictionary

enum class VolunteerStatus(private val code: String, val text: String) {
    ON_WAY("onway", "Выехал"),
    IN_PLACE("inplace", "На месте"),
    LEAVE("leave", "Уехал");

    companion object {
        fun parse(medicine: String): VolunteerStatus {
            return VolunteerStatus.values().firstOrNull { it.code == medicine }
                   ?: VolunteerStatus.ON_WAY
        }
    }
}
